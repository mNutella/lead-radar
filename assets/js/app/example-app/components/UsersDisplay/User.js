import PropTypes from 'prop-types';
import React from 'react';


const User = (props) => {
  const { user } = props;
  return (
    <h6>
      {user.email}
    </h6>
  );
};

User.defaultProps = {
  user: null,
};

User.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  user: PropTypes.any,
};

export default User;
