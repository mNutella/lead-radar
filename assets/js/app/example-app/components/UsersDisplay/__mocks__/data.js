/* eslint-disable camelcase */
import Users from '../../../gql/users.gql';


export const empty = null;

export const no_users = [
  {
    request: {
      query: Users,
      // variables: {
      //   name: 'Buck',
      // },
    },
    result: {
      data: {
        users: [],
      },
    },
  },
];

export const with_one_user = [
  {
    request: { query: Users },
    result: {
      data: {
        users: [
          {
            __typename: 'MyUser',
            email: 'admin@example.com',
          },
        ],
      },
    },
  },
];
