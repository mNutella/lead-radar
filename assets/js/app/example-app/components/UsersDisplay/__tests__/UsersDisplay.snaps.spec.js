/* eslint-disable camelcase */
import { mount, shallow } from 'enzyme';
import React from 'react';
import { MockedProvider } from 'react-apollo/test-utils';
import wait from 'waait';
import User from '../User';
import UsersDisplayMock, { UsersDisplay } from '../UsersDisplay';
import { empty, with_one_user } from '../__mocks__';

describe('<UsersDisplay />', () => {
  test('handles a loading state', () => {
    const wrapper = shallow((
      <UsersDisplay loading />
    ));

    expect(wrapper).toMatchSnapshot();
  });

  test('handles an error state', () => {
    const wrapper = shallow((
      <UsersDisplay error />
    ));

    expect(wrapper).toMatchSnapshot();
  });

  test('markup for empty users', () => {
    const wrapper = shallow((
      <MockedProvider mocks={empty}>
        <UsersDisplay />
      </MockedProvider>
    ));

    expect(wrapper.update().find(UsersDisplay).first().shallow()).toMatchSnapshot();
  });

  test('markup for one user', async () => {
    const wrapper = mount((
      <MockedProvider mocks={with_one_user}>
        <UsersDisplayMock />
      </MockedProvider>
    ));

    await wait(0);

    expect(wrapper.find(UsersDisplay).children()).toMatchSnapshot();
    expect(wrapper.update().find(User).length).toBe(1);
    expect(wrapper.update().find(User).children()).toMatchSnapshot();
  });
});
