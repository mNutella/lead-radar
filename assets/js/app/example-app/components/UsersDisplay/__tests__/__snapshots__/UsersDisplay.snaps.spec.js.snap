// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersDisplay /> handles a loading state 1`] = `
<div>
  Loading
</div>
`;

exports[`<UsersDisplay /> handles an error state 1`] = `
<h1>
  ERROR
</h1>
`;

exports[`<UsersDisplay /> markup for empty users 1`] = `<div />`;

exports[`<UsersDisplay /> markup for one user 1`] = `
<div>
  Loading
</div>
`;

exports[`<UsersDisplay /> markup for one user 2`] = `
<h6>
  admin@example.com
</h6>
`;
